docs/typescript.md

```diff
@@ -0,0 +1,18 @@
+## Frontend Development with Next
+
+The Telescope frontend is currently being developed using Typescript and NextJS. We encourage all new developers to read the [NextJS docs](https://nextjs.org/docs) in order to familiarize themselves with how Next works as well as the [Typescript docs](https://www.typescriptlang.org/docs/) in order to understand how the language operates.
+
+## Design choices
+
+When working on a new component for the Telescope frontend, please consider the following:
+
+- We do not use React.FC/React.FunctionalComponent when making our components. The reason for this can be found [here](https://github.com/facebook/create-react-app/pull/8177).
+- We require all components being created to strictly be exported directly inside the components folder. This is due to the way NextJS processes requests, where we build our pages in the `pages` folder using the components made there.
+- Our style and theme choices can be found [here](./theme-and-colours.md).
```

Comment:
Instead of linking this, let's integrate this document into that file and rename it to `docs/front-end.md`, so all of our front-end notes are in one place.